racket/collects/racket/list.rkt

```diff
@@ -588,6 +590,96 @@
     (vector-set! a j x))
   (vector->list a))
 
+(define (combinations l [k #f])
+  (for/list ([x (in-combinations l k)]) x))
+
+;; Generate combinations of the list `l`.
+;; - If `k` is a natural number, generate all combinations of size `k`.
+;; - If `k` is #f, generate all combinations of any size (powerset of `l`).
+(define (in-combinations l [k #f])
+  (unless (list? l)
+    (raise-argument-error 'in-combinations "list?" 0 l))
+  (when (and k (not (exact-nonnegative-integer? k)))
+    (raise-argument-error 'in-combinations "(or #f exact-nonnegative-integer?)" 1 k))
```

Comment:
Related to the arities comment above. I'd just say `exact-nonnegative-integer?` here. There's no point in ever passing `#f` explicitly.